keylime/keylime_agent.py

```diff
@@ -338,7 +337,7 @@ def initthread():
                         logger.info("No payload script %s found in %s/unzipped", initscript, secdir)
                     else:
                         logger.info("Executing payload script: %s/unzipped/%s", secdir, initscript)
-                        payload_thread = threading.Thread(target=initthread)
+                        payload_thread = threading.Thread(target=initthread, daemon=True)
```

Comment:
I am not sure here. Maybe this is needed and correct, but I feel that making it possible that the Python code can ends and exit before the thread is done, can cause problems.

For example, I am not sure what will happen if the agent throws the payload script in a thread, that is executed inside `{secdir}/zypper`, and we end the agent that maybe will try to umount `{secdir}`. This can make the umount to fail and leave the `{secdir}` present in the system, or cause problems in the payload script later.

Reply:
Yeah you are right. This fix is not totally optimal, but do we want the agent to hang if the payload script does not terminate?